src/lib.rs

```diff
@@ -1,52 +1,42 @@
 use heck::{CamelCase, SnakeCase};
-use itertools::Itertools;
 use proc_macro2::{Ident, Span, TokenStream};
 use quote::{format_ident, quote};
 use serde::{Deserialize, Serialize};
-use serde_json::Value;
 use std::collections::HashMap;
 
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
-struct CommandOption {
-    #[serde(rename = "type")]
-    kind: u8,
+pub struct CommandOption {
+    // TODO: ensure zero is caught as an illegal type for subcommands at runtime,
```

Comment:
could do NonZeroU8 but there are more invalid types than just 0.  Maybe make an enum for it?  you can use #[repr(u8)] and then serde would do the illegal type checks for us